Set configurations Organized configurations in file and envvars

lisandrop opened the pull request:
- All configurations are centralized in AuthConfig. 
- Sensitive data is expected in environment variables and test data will be picked up from test-config.json file. 
-Every parametrized configuration was removed.

warseph commented on src/auth/AuthConfig.ts:
```diff
@@ -0,0 +1,30 @@
+import * as nconf from 'nconf';
```
We are using the [config](https://www.npmjs.com/package/config) module on our packages, it might be a good idea to use the same one.

lisandrop commented:
Reimplemented configurations using config module.
Simplified setup with parameter option.
Added security for missing settings on production. 
Disallow file configuration in production.

a13f932bd652c29a0b224d12ce4f2a465cae5683
